// cloudinary-react-native/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AdvancedImage", {
  enumerable: true,
  get: function () {
    return _AdvancedImage.default;
  }
});
Object.defineProperty(exports, "UploadApiOptions", {
  enumerable: true,
  get: function () {
    return _uploadParams.UploadApiOptions;
  }
});
Object.defineProperty(exports, "explicit", {
  enumerable: true,
  get: function () {
    return _upload.explicit;
  }
});
Object.defineProperty(exports, "rename", {
  enumerable: true,
  get: function () {
    return _upload.rename;
  }
});
Object.defineProperty(exports, "unsignedUpload", {
  enumerable: true,
  get: function () {
    return _upload.unsignedUpload;
  }
});
Object.defineProperty(exports, "upload", {
  enumerable: true,
  get: function () {
    return _upload.upload;
  }
});
Object.defineProperty(exports, "uploadBase64", {
  enumerable: true,
  get: function () {
    return _upload.uploadBase64;
  }
});
var _AdvancedImage = _interopRequireDefault(require("./AdvancedImage"));
var _upload = require("./api/upload");
var _uploadParams = require("./api/upload/model/params/upload-params");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
